main.tf

```diff
@@ -0,0 +1,251 @@
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_s3_bucket" "default" {
+  bucket = "${module.codepipeline_label.id}"
+  acl    = "private"
+  tags   = "${module.codepipeline_label.tags}"
+}
+
+module "codepipeline_assume_label" {
+  source     = "github.com/cloudposse/terraform-terraform-label.git?ref=0.1.2"
+  attributes = ["codepipeline", "assume"]
+  delimiter  = "${var.delimiter}"
+  name       = "${var.name}"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_iam_role" "default" {
+  name               = "${module.codepipeline_assume_label.id}"
+  assume_role_policy = "${data.aws_iam_policy_document.assume.json}"
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    sid = ""
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["codepipeline.amazonaws.com"]
+    }
+
+    effect = "Allow"
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "default" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.default.arn}"
+}
+
+resource "aws_iam_policy" "default" {
+  name   = "${module.codepipeline_label.id}"
+  policy = "${data.aws_iam_policy_document.default.json}"
+}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    sid = ""
+
+    actions = [
+      "ec2:*",
+      "elasticloadbalancing:*",
+      "autoscaling:*",
+      "cloudwatch:*",
+      "s3:*",
+      "sns:*",
+      "cloudformation:*",
+      "rds:*",
+      "sqs:*",
+      "ecs:*",
+      "iam:PassRole",
+    ]
+
+    resources = ["*"]
+    effect    = "Allow"
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "s3" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.s3.arn}"
+}
+
+
+module "codepipeline_s3_policy_label" {
+  source     = "github.com/cloudposse/terraform-terraform-label.git?ref=0.1.2"
+  attributes = ["codepipeline", "s3", "policy"]
+  delimiter  = "${var.delimiter}"
+  name       = "${var.name}"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_iam_policy" "s3" {
+  name   = "${module.codepipeline_s3_policy_label.id}"
+  policy = "${data.aws_iam_policy_document.s3.json}"
+}
+
+data "aws_iam_policy_document" "s3" {
+  statement {
+    sid = ""
+
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion",
+      "s3:GetBucketVersioning",
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.default.arn}",
+      "${aws_s3_bucket.default.arn}/*"
+    ]
+
+    effect = "Allow"
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "codebuild" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.codebuild.arn}"
+}
+
+
+module "codebuild_label" {
+  source     = "github.com/cloudposse/terraform-terraform-label.git?ref=0.1.2"
+  attributes = ["codebuild"]
```

Comment:
Add the incoming attributes to this as well.

Reply:
Like this... https://github.com/cloudposse/terraform-aws-ecs-codepipeline/pull/1/files#diff-7a370d8342e7203b805911c92454f0f4R166